mockgcp/mockcompute/disksv1.go

```diff
@@ -0,0 +1,169 @@
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	id := s.generateID()
+
+	obj := proto.Clone(req.GetDiskResource()).(*pb.Disk)
+	obj.SelfLink = PtrTo("https://compute.googleapis.com/compute/v1/" + name.String())
+	obj.CreationTimestamp = PtrTo(s.nowString())
+	obj.Id = &id
+	obj.Kind = PtrTo("compute#disk")
+	obj.Zone = PtrTo(fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/zones/%s", name.Project.ID, name.Zone))
+	obj.Status = PtrTo("READY")
+	if obj.Type == nil {
+		diskType := "pd-standard"
+		obj.Type = PtrTo(fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/zones/%s/diskTypes/%s", name.Project.ID, name.Zone, diskType))
+	}
+	if obj.PhysicalBlockSizeBytes == nil {
+		obj.PhysicalBlockSizeBytes = PtrTo(int64(4096))
+	}
+
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error creating disk: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+// Updates a Disk resource in the specified project using the data included in the request.
+func (s *DisksV1) Update(ctx context.Context, req *pb.UpdateDiskRequest) (*pb.Operation, error) {
+	reqName := "projects/" + req.GetProject() + "/zones/" + req.GetZone() + "/disks/" + req.GetDisk()
+	name, err := s.parseZonalDiskName(reqName)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+	obj := &pb.Disk{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "disk %q not found", fqn)
+		}
+		return nil, status.Errorf(codes.Internal, "error reading disk: %v", err)
+	}
+
+	// TODO: Implement helper to implement the full rules here
+	proto.Merge(obj, req.GetDiskResource())
+
+	if err := s.storage.Update(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error updating disk: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *DisksV1) Delete(ctx context.Context, req *pb.DeleteDiskRequest) (*pb.Operation, error) {
+	reqName := "projects/" + req.GetProject() + "/zones/" + req.GetZone() + "/disks/" + req.GetDisk()
+	name, err := s.parseZonalDiskName(reqName)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	deleted := (&pb.Disk{})
+	if err := s.storage.Delete(ctx, fqn, deleted); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "disk %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error deleting disk: %v", err)
+		}
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+type zonalDiskName struct {
+	Project *projects.ProjectData
+	Zone    string
+	Name    string
+}
+
+func (n *zonalDiskName) String() string {
+	return "projects/" + n.Project.ID + "/zones/" + n.Zone + "/disks/" + n.Name
+}
+
+// parseZonalDiskName parses a string into a zonalDiskName.
+// The expected form is `projects/*/zones/*/disk/*`.
+func (s *MockService) parseZonalDiskName(name string) (*zonalDiskName, error) {
```

Comment:
In general my philosophy is that rather than adding test coverage in mocks, we should add more tests that drive those mocks harder and check more things (e.g. golden tests)